rollup-interface/src/state_machine/da.rs

```diff
@@ -68,6 +68,27 @@ pub trait DaVerifier {
     ) -> Result<Self::ValidityCondition, Self::Error>;
 }
 
+/// [`AccumulatorStatus`] is a wrapper around an accumulator vector that specifies
+/// whether a [`CountedBufReader`] has finished reading the underlying buffer.
+#[derive(BorshDeserialize, BorshSerialize, Serialize, Deserialize, Debug, Clone, PartialEq)]
+pub enum AccumulatorStatus {
+    /// The underlying buffer has been completely read and [`Vec<u8>`] contains the result
+    Completed(Vec<u8>),
+    /// The underlying buffer still contains elements to be read. [`Vec<u8>`] contains the
+    /// accumulated elements.
+    InProgress(Vec<u8>),
+}
+
+impl AccumulatorStatus {
+    /// Returns the accumulator vector contained in the wrapper
+    pub fn inner_vec(&mut self) -> &mut Vec<u8> {
```

Comment:
I believe this function can return slice, so then we don't need a mutable self, and in the code it is used in `from_slice` anyway.